server/src/main/java/org/apache/iotdb/db/metadata/MManager.java

```diff
@@ -1702,4 +1766,31 @@ public void cacheSchema(String path, MeasurementSchema schema) {
       mRemoteSchemaCache.put(path, schema);
     }
   }
+
+  private void checkMTreeModified() {
+    if (System.currentTimeMillis() - logFile.lastModified() < 60 * 60 * 1000) {
+      logger.info("MTree snapshot is not created because of active modification");
+    } else if (logWriter.getLineNumber() - lastSnapshotLogLineNumber < mtreeSnapshotInterval) {
+      logger.info("MTree snapshot need not be created");
+    } else {
+      lock.writeLock().lock();
+      logger.info("Start creating MTree snapshot. This may take a while...");
+      try {
+        mtree.serializeTo(mtreeSnapshotTmpPath, logWriter.getLineNumber());
+        lastSnapshotLogLineNumber = logWriter.getLineNumber();
+        File tmpFile = SystemFileFactory.INSTANCE.getFile(mtreeSnapshotTmpPath);
+        File snapshotFile = SystemFileFactory.INSTANCE.getFile(mtreeSnapshotPath);
+        if (snapshotFile.exists()) {
+          Files.delete(snapshotFile.toPath());
+        }
+        if (tmpFile.renameTo(snapshotFile)) {
+          logger.info("Finish creating MTree snapshot to {}.", mtreeSnapshotPath);
+        }
+      } catch (IOException e) {
+        logger.warn("Failed to create MTree snapshot to {}", mtreeSnapshotPath, e);
```

Comment:
remove tmp file

Reply:
No need. It will be removed when restarting